Test/Test/Source.cpp

```diff
@@ -1,78 +1,118 @@
 #include <iostream>
 #include <fstream>
 #include <string>
+#include <stdio.h>
+
+
 #include "ConsoleReader.h"
 
+#include <log4cpp/Portability.hh>
+#include <log4cpp/PropertyConfigurator.hh>
+#include <log4cpp/Category.hh>
+#include <log4cpp/Appender.hh>
+#include <log4cpp/FileAppender.hh>
+#include <log4cpp/OstreamAppender.hh>
+#include <log4cpp/Layout.hh>
+#include <log4cpp/BasicLayout.hh>
+#include <log4cpp/Priority.hh>
+#include <log4cpp/NDC.hh>
+
+
 const int possibleChars = 128;
 
-void calculateOccurs(int arr[], std::string line) {
-	for (int j = 0; j < line.length(); ++j) {
-		++arr[int(line[j])];
+void calculateOccurs(int charOccurs[], std::string const &inputLine) {
+	for (int j = 0; j < inputLine.length(); ++j) {
+		++charOccurs[int(inputLine[j])];
 	}
 }
 
 
-bool calculateOccursInFile(std::string path, int arr[]) {
-	std::string line;
+bool calculateOccursInFile(std::string const &inputFilepath, int charOccurs[]) {
+	std::string inputLine;
 	std::ifstream input;
-	input.open(path);
+	input.open(inputFilepath);
 	if (!input.is_open()) {
-		std::cout << "Could not open the file with a path " << path;
 		return false;
 	}
 	while (!input.eof()) {
-		getline(input, line);
-		// std::cout << line << std::endl;
+		getline(input, inputLine);
 
 		// calculates the number of occurrences of each unique symbol
-		calculateOccurs(arr, line);
+		calculateOccurs(charOccurs, inputLine);
 	}
 	input.close();
 	return true;
 }
-// so this is not really reusable, and I'm not sure if that can be fixed, or if it even is necessary
 
 
-void outputToFile(std::string outputPath, int arr[]) {
+bool outputToFile(std::string const &outputFilePath, int charOccurs[]) {
 	std::ofstream output;
-	output.open(outputPath);
+	output.open(outputFilePath);
+	if (!output.is_open()) {
+		return false;
+	}
+
 
-	std::cout << "Creating output.txt file with char occurences in input file\n";
 	for (int i = 0; i < possibleChars; ++i) {
-		if (arr[i]) {
-			// std::cout << char(i) << ": " << charOccurs[i] << std::endl;
-			output << char(i) << ": " << arr[i] << std::endl;
+		if (charOccurs[i]) {
+			output << char(i) << ": " << charOccurs[i] << std::endl;
 		}
 	}
 	output.close();
+	return true;
 }
 
 
 int main() {
+	// Logging stuff
+	std::string initFileName = "log4cpp.properties";
+	log4cpp::PropertyConfigurator::configure(initFileName);
+
+	log4cpp::Category& root = log4cpp::Category::getRoot();
```

Comment:
JFYI. You may use one root logger for all the application. But, sure you may have several different loggers to make it easier to configure them separately.